graph/patterns/change-status.md

```diff
@@ -0,0 +1,107 @@
+# Change Status & Action Request
+
+Microsoft Graph API Design Pattern
+
+_The change status pattern is an effective alternative to model the ability to trigger side effects while keeping an agnostic, data driven approach_
+
+## Problem
+
+In some cases, exposed resources can trigger side effects that modify the system behavior. For instance, a [`riskyUser`](https://docs.microsoft.com/en-us/graph/api/resources/riskyuser?view=graph-rest-1.0) can be [`confirmedCompromised`](https://docs.microsoft.com/en-us/graph/api/riskyuser-confirmcompromised?view=graph-rest-1.0&tabs=http) or [`dismissed`](https://docs.microsoft.com/en-us/graph/api/riskyuser-confirmcompromised?view=graph-rest-1.0&tabs=http).
+
+These situation are often modeled as a `POST` request to a specific endpoint in the same URI space of the resource, triggering the side effect directly and not returning any data. While it might be a quick and easy solution, there are some considerations:
+
+- The API should be exposing resources and allow their manipulation, while Actions resemble a remote procedure call on an object, typical of the [OOP][1]
+- It couples the request with the side effect, making it more difficult to test, log, replay and reason about what is happening in the system
+- It forces the consumer to follow the [OOP][1] approach at the API level. We do not want to pursue any specific style and the developers writing the client might be working in other ways
```

Comment:
If we want to refer to a particular style it should be an API style which is not necessarily a language style

Reply:
I am not too sure I am following your point.

What I am trying to communicate here is that the current way we are modeling this situation is forcing the SDK to also be modeled in such way - while there are developers working in different manners.

One of the advantages of representing status changes through data is that the SDK is more agnostic and does not clash with anybody.

Let me know whether this makes sense or not